web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2016 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define('web_app_drawer', function(require) {
+    'use strict';
+    
+    var $ = require('$');
+    var Menu = require('web.Menu');
+    var Class = require('web.Class');
+    var SearchView = require('web.SearchView');
+    var core = require('web.core');
+    
+    Menu.include({
+        // Force all_outside to prevent app icons from going into more menu
+        reflow: function() {
+            this._super('all_outside');
+        },
+    });
+    
+    SearchView.include({
+        
+        // Prevent focus of search field on mobile devices
+        toggle_visibility: function (is_visible) {
+             $('div.oe_searchview_input').last()
+                .one('focus', $.proxy(this.preventMobileFocus, this));
+            return this._super(is_visible);
+        },
+        
+        // It prevents focusing of search el on mobile
+        preventMobileFocus: function(event) {
+            if (this.isMobile()) {
+                event.preventDefault();
+            }
+        },
+        
+        // For lack of Modernizr, TouchEvent will do
+        isMobile: function () {
+            try{
+                document.createEvent('TouchEvent');
+                return true;
```

Comment:
It would be nice to be able to drag the drawer from left border like any other android app